### ISSUE TITLE
System.QueryException: No such column 'CurrencyIsoCode' on entity 'OpportunityContactRole' while using fflib library

### ISSUE THREAD

**Issue by meta-arun-parmar** [apex-enterprise-patterns/fflib-apex-common#304]
I am using fflib library and querying OpportunityContactRole object in trigger. But its throwing below error -> Opportunity_Trigger: execution of BeforeUpdate caused by: System.QueryException: No such column 'CurrencyIsoCode' on entity 'OpportunityContactRole'. If you are attempting to use a custom field, be sure to append the '__c' after the custom field name. Please reference your WSDL or the describe call for the appropriate names.

there is no CurrencyIsoCode field on OpportunityContactRole object, The currency field is added by fllib. can anyone please help me on this ?

**Comment by wimvelzeboer** [comment]
Can you share the selector class that is throwing the error?

In the class `fflib_SObjectSelector` (line 40) it is checking if the field is available on the object: `describeWrapper.getFieldsMap().keySet().contains('currencyisocode');`.

Only when multicurrency is enabled and the above line results to true, the CurrencyIsoCode will de added
(line 386:)
```
if(UserInfo.isMultiCurrencyOrganization() && CURRENCY_ISO_CODE_ENABLED)
            queryFactory.selectField(relationshipFieldPath+'.CurrencyIsoCode');		
```

You can try to debug these lines and see where they result to, that might give you an idea what is going wrong.

**Comment by meta-arun-parmar** [comment]
Hi wimvelzeboer,

`UserInfo.isMultiCurrencyOrganization()` is true and `CURRENCY_ISO_CODE_ENABLED `is also true.
I am not sure why `CURRENCY_ISO_CODE_ENABLED` is true, because when I check in setup->Object Manager-> OpportunityContactRole, then there is no CurrencyIsoCode field. But still I'm getting this field in `describeWrapper.getFieldsMap().keySet().contains('currencyisocode');`

**Comment by wimvelzeboer** [comment]
If you can share a code snippet replicating the issue, a sandbox or scratch org then I can have a closer look to the origin of your issue.

**Comment by meta-arun-parmar** [comment]
```
public class SLT_OpportunityContactRoles extends fflib_SObjectSelector{
	
    public List<Schema.SObjectField> getSObjectFieldList() {
        return new List<Schema.SObjectField> {OpportunityContactRole.Id, OpportunityContactRole.OpportunityId, OpportunityContactRole.isPrimary};
    }
	
    public Schema.SObjectType getSObjectType() {
        return OpportunityContactRole.sObjectType;
    }
	
    public Map<Id, OpportunityContactRole> selectByOpportunityId(Set<ID> idSet) {
        return new Map<Id, OpportunityContactRole>((List<OpportunityContactRole>) Database.query(newQueryFactory().setCondition('OpportunityId in :idSet').toSOQL()));
    }
}
```

Getting error when call method selectByOpportunityId. I'm getting this issue in latest fflib package(fflib class version is 47.0).

**Comment by wimvelzeboer** [comment]
@meta-arun-parmar I created a Scratch Org, deployed fflib-apex-mocks and fflib-apex-commons with your code snippet.
Then created some test data and run this in anonymous apex
`Map<Id, OpportunityContactRole> opportunityContactRolesByIds = 
    new SLT_OpportunityContactRoles()
        .selectByOpportunityId(new Set<Id> {'0063D00000EPkK4QAL'});
System.debug(opportunityContactRolesByIds);`

Where the Id is the Id of the opportunity.
It just runs successfully without errors:

> 03:31:22.174 (227652707)|USER_DEBUG|[21]|DEBUG|SELECT CurrencyIsoCode, Id, IsPrimary, OpportunityId FROM OpportunityContactRole WHERE OpportunityId in :idSet ORDER BY CreatedDate ASC NULLS FIRST
> 03:31:22.174 (246692313)|USER_DEBUG|[2]|DEBUG|{00K3D000003dpPFUAY=OpportunityContactRole:{CurrencyIsoCode=USD, Id=00K3D000003dpPFUAY, IsPrimary=false, OpportunityId=0063D00000EPkK4QAL}}

I think there might be something else wrong in your environment, causing the issue. Maybe in the Opportunity_Trigger ?

**Comment by meta-arun-parmar** [comment]
@wimvelzeboer , Thank you so much for your support. You were right. I was calling SLT_OpportunityContactRoles class from Opportunity.  SLT_OpportunityContactRoles has version 40.0 , so i was getting this error, im getting the same error till version 46.0.
Its working proper in version 47.0. 

For your clarification you can change this class version to 40 and test, you should face the same error.

Thanks for your support, i appreciate that.

**Comment by wimvelzeboer** [comment]
@meta-arun-parmar The error is occurring indeed when I change the selector class version into 40. 

Your error was caused by a mismatch in API versions of the two different classes. `fflib_SObjectSelector` runs in API 47.0 which is checking if the field is available, which it is in API 47.0. But your query was running in API 40.0 where that field isn't available. 

I think we can close this issue and mark it as resolved.
